bin/crew

```diff
@@ -646,18 +646,25 @@ def update
       puts "Package file for #{package[:name]} not found. :(".lightred if @opt_verbose
       next
     end
-    if package[:binary_sha256] && package[:binary_sha256] != @pkg.get_binary_sha256(@device[:architecture])
-      canBeUpdated += 1
-      puts "#{@pkg.name} could be updated (rebuild)"
-    end
-    if package[:version] != @pkg.version
-      canBeUpdated += 1
+    differentVersion = (package[:version] != @pkg.version)
+    hasSha = @pkg.get_binary_sha256(@device[:architecture]).to_s.empty? || package[:binary_sha256].to_s.empty? ? false : true
+    differentSha = (hasSha && package[:binary_sha256] != @pkg.get_binary_sha256(@device[:architecture]))
+
+    canBeUpdated += 1 if differentVersion || differentSha
+
+    if differentVersion
       puts "#{@pkg.name} could be updated from #{package[:version]} to #{@pkg.version}"
+    elsif !differentVersion && differentSha
+      puts "#{@pkg.name} could be updated (rebuild)"
+    elsif differentVersion && !differentSha && hasSha
+      canBeUpdated -= 1
+      puts "#{@pkg.name} has a version change but does not have updated binaries".yellow
```

Comment:
This condition is not possible since the first condition passes.  What you likely want is:
```
    if differentVersion
      puts "#{@pkg.name} could be updated from #{package[:version]} to #{@pkg.version}"
    elsif differentSha
      puts "#{@pkg.name} could be updated (rebuild)"
    end
```

Reply:
Fixed.